main.tf

```diff
@@ -17,15 +19,50 @@ locals {
   )
 
   availability_zone = var.multi_az ? null : var.availability_zone
+
+  create_user     = local.enabled && length(var.database_user) == 0
+  create_password = local.enabled && length(var.database_password) == 0
+
+  database_user     = local.create_user ? substr(join("", random_pet.database_user.*.id), 0, 16) : var.database_user
+  database_password = local.create_password ? join("", random_password.database_password.*.result) : var.database_password
+}
+
+resource "random_pet" "database_user" {
+  count = local.create_user ? 1 : 0
+
+  # word length
+  length = 5
+
+  separator = ""
+
+  keepers = {
+    db_name = var.database_name
+  }
+}
+
+resource "random_password" "database_password" {
+  count = local.create_password ? 1 : 0
+
+  # character length
+  length = 33
+
+  # Leave special characters out to avoid quoting and other issues.
+  # Special characters have no additional security compared to increasing length.
+  special          = false
+  override_special = "!#$%^&*()<>-_"
```

Comment:
AFAIU, y`ou don't need `override_special` if you include `special = false`. No harm no foul, but just mentioning.